cookiecutter.json

```diff
@@ -1,23 +1,20 @@
 {
     "project_name": "ProjectName",
-    "repo_name": "{{ cookiecutter.project_name.lower().replace(' ', '_') }}",
-    "first_module_name": "{{ cookiecutter.repo_name.lower().replace(' ', '_') }}",
+    "repo_name": "{{ cookiecutter.project_name.lower().replace(' ', '_').replace('-', '_').replace(' ', '_') }}",
+    "github_username": "Your account name or organization name on GitHub where repo will be hosted",
```

Comment:
This is a bit ambiguous: so for something hosted under the MDAnalysis org I would say "MDAnalysis" but if it were hosted under my personal account it would be "orbeckst"?

When a variable is called "github_username" I'd always answer "orbeckst".

Perhaps it would be worthwhile distinguishing the two uses of "username" more clearly, e.g. github_id = orbeckst and github_repository_toplevel = MDAnalysis | orbeckst (not sure what the best description for the "toplevel" is).

Reply:
I don't know how easy it is to customize the process but one could ask

1. github ID? <-- orbeckst
2. hosting under your personal account Y/N
    1. yes : nothing to do (automatically set toplevel <-- github ID)
    2. no: organization? <-- MDAnalysis

and then use this information to build the repo URL

Reply:
This is, unfortunately, non-trivial to support -- `cookiecutter` doesn't support conditional questions, or much logic in the question asking process. It also doesn't allow for both default values *and* helpful prompts. I pushed a trial change that looks like this:

```
$ cookiecutter .
project_name [ProjectName]: asdf
repo_name [asdf]:
github_username [Your personal GitHub username]: lilyminium
github_host_account [GitHub account for hosting asdf (e.g. lilyminium or an organization account). Press Enter to use lilyminium]:
first_module_name [asdf]:
author_name [Your name (or your organization/company/team)]: lily
author_email [Your email (or your organization/company/team)]: l@g.com
description [A short description of the project.]:
Select dependency_source:
1 - Prefer conda-forge over the default anaconda channel with pip fallback
2 - Prefer default anaconda channel with pip fallback
3 - Dependencies from pip only (no conda)
Choose from 1, 2, 3 [1]:
Select include_ReadTheDocs:
1 - y
2 - n
Choose from 1, 2 [1]:
```

and the default answer to `github_host_account` is edited with Jinja in the docstring of `hooks/pre_gen_project.py`. It's hacky but works...

I should mention that all these individual PRs have been working towards a main cohesive cookiecutter that I've got up at https://github.com/lilyminium/proto-cookiecutter-mda if you want to trial going through the options yourself:

```
cookiecutter https://github.com/lilyminium/proto-cookiecutter-mda
```